Improvements related to save of config file by LightningCLI (#7963)
PyTorchLightning/pytorch-lightning: CHANGELOG.md

- Added `on_load_checkpoint` and `on_save_checkpoint` hooks to the `PrecisionPlugin` base class ([#7831](https://github.com/PyTorchLightning/pytorch-lightning/pull/7831))

> - LightningCLI now aborts with a clearer message if config already exists and disables save config for fast_dev_run!=False ([#7963](https://github.com/PyTorchLightning/pytorch-lightning/pull/7963))

Reviewer: ```suggestion
- `LightningCLI` now aborts with a clearer message if config already exists and disables save config during `fast_dev_run`([#7963](https://github.com/PyTorchLightning/pytorch-lightning/pull/7963))
```

> - LightningCLI now does not aborts with a clearer message if config already exists and disables save config for fast_dev_run!=False ([#7859](https://github.com/PyTorchLightning/pytorch-lightning/discussions/7859))

Reviewer 2: ```suggestion
- LightningCLI now aborts with a clearer message if config already exists and disables save config for fast_dev_run!=False ([#7859](https://github.com/PyTorchLightning/pytorch-lightning/discussions/7859))
```
